src/Cli/dotnet/Commands/Tool/Execute/ToolExecuteCommand.cs

```diff
@@ -0,0 +1,181 @@
+                nugetConfig: _configFile != null ? new(_configFile) : null,
+                sourceFeedOverrides: _sources,
+                additionalFeeds: _addSource);
+
+        (var bestVersion, var packageSource) = _toolPackageDownloader.GetNuGetVersion(packageLocation, packageId, _verbosity, versionRange, _restoreActionConfig);
+
+        IToolPackage toolPackage;
+
+        //  TargetFramework is null, which means to use the current framework.  Global tools can override the target framework to use (or select assets for),
+        //  but we don't support this for local or one-shot tools.
+        if (!_toolPackageDownloader.TryGetDownloadedTool(packageId, bestVersion, targetFramework: null, out toolPackage))
+        {
+            if (!UserAgreedToRunFromSource(packageId, bestVersion, packageSource))
+            {
+                if (_interactive)
+                {
+                    Reporter.Error.WriteLine(CliCommandStrings.ToolDownloadCanceled.Red().Bold());
+                    return ERROR_CANCELLED;
+                }
+                else
+                {
+                    Reporter.Error.WriteLine(CliCommandStrings.ToolDownloadNeedsConfirmation.Red().Bold());
+                    return 1;
+                }
+            }
+
+            //  We've already determined which source we will use and displayed that in a confirmation message to the user.
+            //  So set the package location here to override the source feeds to just the source we already resolved to.
+            //  This does mean that we won't work with feeds that have a primary package but where the RID-specific packages are on
+            //  other feeds, but this is probably OK.
+            var downloadPackageLocation = new PackageLocation(
+                nugetConfig: _configFile != null ? new(_configFile) : null,
+                sourceFeedOverrides: [packageSource.Source],
+                additionalFeeds: _addSource);
+
+            toolPackage = _toolPackageDownloader.InstallPackage(
+                downloadPackageLocation,
+                packageId: packageId,
+                verbosity: _verbosity,
+                versionRange: new VersionRange(bestVersion, true, bestVersion, true),
+                isGlobalToolRollForward: false,
+                restoreActionConfig: _restoreActionConfig);
+        }
+
+        var commandSpec = ToolCommandSpecCreator.CreateToolCommandSpec(toolPackage.Command.Name.Value, toolPackage.Command.Executable.Value, toolPackage.Command.Runner, _allowRollForward, _forwardArguments);
+        var command = CommandFactoryUsingResolver.Create(commandSpec);
+        var result = command.Execute();
+        return result.ExitCode;
+    }
+
+    private bool UserAgreedToRunFromSource(PackageId packageId, NuGetVersion version, PackageSource source)
+    {
+        if (_yes)
+        {
+            return true;
+        }
+
+        if (!_interactive)
+        {
+            return false;
```

Comment:
Good call to default to 'no' in this instance!

Reply:
Ah, that actually makes sense. No is better than no default, but I think we should change it too.